### ISSUE TITLE
Proper formatting for DateTime Index

### ISSUE THREAD

**Issue by ParthivNaresh** [alteryx/evalml#2134]
As an extension to #2124, a warning can be included that assesses whether the date has been properly formatted. It might be useful to have certain default formats for dates that are passed in, such as `mm/dd/YYYY %H:%M:%S`.

For example, in some cases, Pandas will infer the year 55 as 2055 and then 77 as 1977. This would entirely misconstrue the timeline of data. Users should be encouraged to format their datetime column/index to include the full year.

**Comment by dsherry** [comment]
Could this be a woodwork issue? @gsheni @thehomebrewnerd

**Comment by gsheni** [comment]
@dsherry @ParthivNaresh Would a Woodwork solution throw a warning/error if the datetime column is not properly formatted? Seems that we need a validation check for the Datetime Logical Type.

**Comment by thehomebrewnerd** [comment]
Just to be clear on the current Woodwork behavior: If a column is specified to have a logical type of `Datetime`, Woodwork will confirm that this column contains valid datetime information by attempting to convert the series using a call to `pd.to_datetime`. If the data cannot be converted an error is raised.

For datetimes that might contain non-standard formatting, Woodwork users can create an instance of the Datetime logical type and specify the formatting string. If specified, this formatting string will be used in the call to `pd.to_datetime`.

So in short, if data can be converted through a call to `pd.to_datetime`, either with or without a formatting string, Woodwork will recognize this as a valid `Datetime` column, otherwise and error will be raised.

**Comment by dsherry** [comment]
@thehomebrewnerd thanks for that context. Good to know the current behavior.

@gsheni yep I agree. I think that it would be great if ww had an API for specifying datetime parse strings and throwing helpful errors when a string doesn't match the formatting. That would allow EvalML/FT users to use woodwork to catch and fix these problems before moving on.

Is there a woodwork issue for this yet? If so, let's close this.

**Comment by thehomebrewnerd** [comment]
> @gsheni yep I agree. I think that it would be great if ww had an API for specifying datetime parse strings and throwing helpful errors when a string doesn't match the formatting. That would allow EvalML/FT users to use woodwork to catch and fix these problems before moving on.

@dsherry I think this behavior already exists in Woodwork, with possibly a debate about how "helpful" the error is. Take this for example:
```python
import pandas as pd
import woodwork as ww
from woodwork.logical_types import Datetime

df = pd.DataFrame({
        'index': [0, 1, 2],
        'dates': ["2019/01/01", "2019/01/02", "2019/01/03"],
        'ymd_funky': ["2019~01~01", "2019~01~02", "2019~01~03"],
})

df.ww.init(
    name='df_name',
    logical_types={
        'dates': Datetime,
        'ymd_funky': Datetime,
    }
)
```

If you run the code block above, trying to initialize both datetime columns with just the standard `Datetime` logical type, you will get a Woodwork `TypeConversionError` with this message:
```
TypeConversionError: Error converting datatype for ymd_special from type object to type datetime64[ns]. Please confirm the underlying data is consistent with logical type Datetime.
```
Also within the stack trace is this message from pandas:
```
ParserError: Unknown string format: 2019~01~01
```

Now, if you try this again, except specifying the datetime format of the `ymd_funky` column, everything works fine with no errors:
```python
df.ww.init(
    name='df_name',
    logical_types={
        'dates': Datetime,
        'ymd_funky': Datetime(datetime_format='%Y~%m~%d'),
    }
)
df.ww
```

Output:
```
            Physical Type Logical Type Semantic Tag(s)
Column                                                
index               int64      Integer     ['numeric']
dates      datetime64[ns]     Datetime              []
ymd_funky  datetime64[ns]     Datetime              []
```

Finally, if you try to use the `ymd_funky` format string on the more standard formatted `dates` column this the error you get:
```python
df.ww.init(
    name='df_name',
    logical_types={
        'dates': Datetime(datetime_format='%Y~%m~%d'),
        'ymd_funky': Datetime(datetime_format='%Y~%m~%d'),
    }
)
```
```
TypeConversionError: Error converting datatype for dates from type object to type datetime64[ns]. Please confirm the underlying data is consistent with logical type Datetime.
```

and within the stack trace there is this from pandas:
```
ValueError: time data '2019/01/01' does not match format '%Y~%m~%d' (match)
```

As of now, there is not a Woodwork issue related to this, but if you think there is something in the current behavior that needs to be updated, we can certainly create one.

**Comment by dsherry** [comment]
@thehomebrewnerd thank you for that context. Good to know we have that support in woodwork currently.

I will close this and file a separate issue in woodwork to track a "smart inference" feature.
